types/ember/index.d.ts

```diff
@@ -1173,7 +1174,7 @@ declare module 'ember' {
              * can pass an optional second argument with the target value. Otherwise
              * this will match any property that evaluates to `true`.
              */
-            findBy(key: string, value: string): T | undefined;
+            findBy(key: string, value: any): T | undefined;
```

Comment:
Are the docs wrong here? They say this is of type `string`: [method doc](https://emberjs.com/api/ember/2.16/classes/Enumerable/methods/findBy?anchor=findBy).

Reply:
Follow-up: the docs appear to be correct; [`findBy` delegates to the `iter` function](https://github.com/emberjs/ember.js/blob/v2.16.2/packages/ember-runtime/lib/mixins/enumerable.js#L550-L552), which [does a lookup with a key](https://github.com/emberjs/ember.js/blob/v2.16.2/packages/ember-runtime/lib/mixins/enumerable.js#L48-L54).

Reply:
I'm confused too now!  As I understand it, findBy

1. Iterates through the collection
2. Uses `key` to retrieve a value from the collection
3. if a `value` is provided the two are compared using `===`, so value could be of `any` type

So, I think the type should actually be `value?: any`

example

![image](https://user-images.githubusercontent.com/1583673/34426471-e107c6b8-ec71-11e7-9925-7cff684ad677.png)